src/stores/OwnProfileStore.ts

```diff
@@ -115,6 +124,8 @@ export class OwnProfileStore extends AsyncStoreWithClient<IState> {
         // We specifically do not use the User object we stored for profile info as it
         // could easily be wrong (such as per-room instead of global profile).
         const profileInfo = await this.matrixClient.getProfileInfo(this.matrixClient.getUserId());
+        if (profileInfo.displayname) window.localStorage.setItem(KEY_DISPLAY_NAME, profileInfo.displayname);
+        if (profileInfo.avatar_url) window.localStorage.setItem(KEY_AVATAR_URL, profileInfo.avatar_url);
```

Comment:
we should also probably clear the values if they've been unset (remove avatar, etc)

Reply:
oh, good point